src/hotspot/src/share/vm/opto/type.cpp

```diff
@@ -2552,13 +2552,19 @@ TypeOopPtr::TypeOopPtr(TYPES t, PTR ptr, ciKlass* k, bool xk, ciObject* o, int o
         } else if (klass() == ciEnv::current()->Class_klass() &&
                    _offset >= InstanceMirrorKlass::offset_of_static_fields()) {
           // Static fields
-          assert(o != NULL, "must be constant");
-          ciInstanceKlass* k = o->as_instance()->java_lang_Class_klass()->as_instance_klass();
-          ciField* field = k->get_field_by_offset(_offset, true);
-          assert(field != NULL, "missing field");
-          BasicType basic_elem_type = field->layout_type();
-          _is_ptr_to_narrowoop = UseCompressedOops && (basic_elem_type == T_OBJECT ||
+          ciField* field = NULL;
+          if (const_oop() != NULL) {
```

Comment:
I'm not sure about this line and the next one - is it possible for the value of `const_oop()` to have changed between these 2 usages? How expensive is the call? I'd argue we should instead do something like `oop = const_oop(); if (oop != NULL) { ... }`

Reply:
This was adapted from upstream and is the same as 17: https://github.com/corretto/corretto-17/blob/d245e8d15be8f906dba27353427848c3e24eff10/src/hotspot/share/opto/type.cpp#L3136

Reply:
Why cherry-pick this change instead of backporting 8255466?

Reply:
Not sure why I did that, let me backport